CONTRIBUTING.rst

```diff
@@ -54,51 +54,394 @@ If you are proposing a feature:
+    git clone git@github.com:yourusername/ACT.git
+
+After that, from within the ACT directory, do::
+
+    git remote add upstream https://github.com/ANL-DIGR/ACT.git
+
+Install
+-------
+
+The easiest method for using ACT and is dependencies is by using:
+`Anaconda <https://www.anaconda.com/download/#>`_ or
+`Miniconda <https://conda.io/miniconda.html>`_.
+From within the ACT directory, you can use::
+
+    pip install -e .
+
+This downloads ACT in development mode. Do this preferably in a conda
+environment. For more on Anaconda and environments:
+
+- https://anl-digr.github.io/ACT/CREATING_ENVIRONMENTS.html
+
+Working with Git Branches
+-------------------------
+
+When contributing to ACT, the changes created should be in a new branch
+under your forked repository. Let's say the user is adding a new plot display.
+Instead of creating that new function in your master branch. Create a new
+branch called ‘wind_rose_plot’. If everything checks out and the admin
+accepts the pull request, you can then merge the master branch and
+wind_rose_plot branch.
+
+To delete a branch both locally and remotely, if done with it::
+
+                git push origin --delete <branch_name>
+                git branch -d <branch_name>
+
+or in this case::
+                
+                git push origin --delete wind_rose_plot
+                git branch -d wind_rose_plot
+
+
+To create a new branch::
+
+                git checkout -b <branch_name>
+
+If you have a branch with changes that have not been added to a pull request
+but you would like to start a new branch with a different task in mind. It
+is recommended that your new branch is based on your master. First::
+
+                git checkout master
+
+Then::
+
+                git checkout -b <branch_name>
+
+This way, your new branch is not a combination of your other task branch and
+the new task branch, but is based on the original master branch.
+
+Typing `git status` will not only inform the user of what files have been
+modified and untracked, it will also inform the user of which branch they
+are currently on.
+
+To switch between branches, simply type::
+
+                git checkout <branch_name>
+
+Python File Setup
+-----------------
+
+In a new .py file, the top of the code should have the function or class
+location, sphinx comments for template configuration, and the public and
+private functions and classes within the .py file. Public functions and
+classes are listed first and then private functions and classes. Private
+functions and classes should have a underscore in front of the name. A space
+is needed between the last function or class and the closing docstring
+quotation.
+
+An example:
+
+.. code-block:: python
+
+        """
+        act.plotting.TimeSeriesDisplay
+        ==============================
+        Class for creating timeseries plots from ACT datasets.
+
+        .. autosummary::
+            :toctree: generated/
+
+            TimeSeriesDisplay
```

Comment:
I would use a different doc string here? I think it's trying to import an RST file called TimeSeriesDisplay for some reason.